File: VB/Solution1.Module/BusinessObjects/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace Solution1.Module.BusinessObjects
	<DefaultClassOptions> _
	Public Class Note
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here or place it only when the IsLoading property is false:
			' if (!IsLoading){
			'    It is now OK to place your initialization code here.
			' }
			' or as an alternative, move your initialization code into the AfterConstruction method.
		End Sub
		Public Property Name() As String
			Get
				Return GetPropertyValue(Of String)("Name")
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Name", value)
			End Set
		End Property
		Public Property IsProcessed() As Boolean
			Get
				Return GetPropertyValue(Of Boolean)("IsProcessed")
			End Get
			Set(ByVal value As Boolean)
				SetPropertyValue(Of Boolean)("IsProcessed", value)
			End Set
		End Property
	End Class

	<DefaultClassOptions> _
	Public Class Task
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here or place it only when the IsLoading property is false:
			' if (!IsLoading){
			'    It is now OK to place your initialization code here.
			' }
			' or as an alternative, move your initialization code into the AfterConstruction method.
		End Sub
		Public Property CreatedOn() As DateTime
			Get
				Return GetPropertyValue(Of DateTime)("CreatedOn")
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue(Of DateTime)("CreatedOn", value)
			End Set
		End Property
		Public Property Name() As String
			Get
				Return GetPropertyValue(Of String)("Name")
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Name", value)
			End Set
		End Property
		Public Property SourceNote() As Note
			Get
				Return GetPropertyValue(Of Note)("SourceNote")
			End Get
			Set(ByVal value As Note)
				SetPropertyValue(Of Note)("SourceNote", value)
			End Set
		End Property
		Public Property TaskType() As TaskType
			Get
				Return GetPropertyValue(Of TaskType)("TaskType")
			End Get
			Set(ByVal value As TaskType)
				SetPropertyValue(Of TaskType)("TaskType", value)
			End Set
		End Property
	End Class

	<DefaultClassOptions> _
	Public Class TaskType
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Name() As String
			Get
				Return GetPropertyValue(Of String)("Name")
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Name", value)
			End Set
		End Property
	End Class
End Namespace

File: CS/Solution1.Module/BusinessObjects/DomainObject1.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace Solution1.Module.BusinessObjects {
    [DefaultClassOptions]
    public class Note : BaseObject {
        public Note(Session session)
            : base(session) {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here or place it only when the IsLoading property is false:
            // if (!IsLoading){
            //    It is now OK to place your initialization code here.
            // }
            // or as an alternative, move your initialization code into the AfterConstruction method.
        }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
        public bool IsProcessed {
            get { return GetPropertyValue<bool>("IsProcessed"); }
            set { SetPropertyValue<bool>("IsProcessed", value); }
        }
    }

    [DefaultClassOptions]
    public class Task : BaseObject {
        public Task(Session session)
            : base(session) {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here or place it only when the IsLoading property is false:
            // if (!IsLoading){
            //    It is now OK to place your initialization code here.
            // }
            // or as an alternative, move your initialization code into the AfterConstruction method.
        }
        public DateTime CreatedOn {
            get { return GetPropertyValue<DateTime>("CreatedOn"); }
            set { SetPropertyValue<DateTime>("CreatedOn", value); }
        }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
        public Note SourceNote {
            get { return GetPropertyValue<Note>("SourceNote"); }
            set { SetPropertyValue<Note>("SourceNote", value); }
        }
        public TaskType TaskType {
            get { return GetPropertyValue<TaskType>("TaskType"); }
            set { SetPropertyValue<TaskType>("TaskType", value); }
        }        
    }

    [DefaultClassOptions]
    public class TaskType : BaseObject {
        public TaskType(Session session)
            : base(session) {
        }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
    }
}
